### ISSUE TITLE
need better description about the `amount` field in search key

### ISSUE THREAD

**Issue by cryptowen** [nervosnetwork/ckb-indexer#29]
We came across a bug today due to a misunderstanding of the amount field in search key. 
The situation description:
- We use `res.result.objects.length < length-in-search-key` as a condition to stop the search.
- We came across a case that it returns empty objects for an existing cell when we search with typescript as parameter and lockscript in filter. The amount is 0x64. When we set the amount to 0x6400, the result meets expect.

We guess that:
- The amount in search key is the amount searched before filter.
- So when we use filter in the search key, we can not use `res.result.objects.length < length-in-search-key` as stop condition.  
- The best practice for stop condition may be searching with last_cursor until it becomes `0x`.

We want to confirm if our guess is right. And if so, it's better to document it more clearly.

**Comment by quake** [comment]
There are no `amount` field in the search key, do you mean the `output_capacity_range`? could you give a rpc request parameter and return result example?

**Comment by cryptowen** [comment]
sorry, not `amount`, the `limit` in search key.

**Comment by cryptowen** [comment]
example:

```
echo '{
  "id": 2,
  "jsonrpc": "2.0",
  "method": "get_cells",
  "params": [
      {
          "filter": {
            "script": {
              "code_hash": "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
              "hash_type": "type",
              "args": "0x838e79dcef9cbf819a32778c8cfcc81bb2555561"
            }
          },
          "script": {
            "code_hash": "0xc5e5dcf215925f7ef4dfaf5f4b4f105bc321c02776d6e7d52a1db3fcd9d011a4",
            "hash_type": "type",
            "args": "0x4580e3fd3a6623eb26f229239286cee63e18bcafb38bc6c5d0de5a8c587647c2"
          },
          "script_type": "type"
      },
      "asc",
      "0x64"
  ]
}' | tr -d '\n' | curl -H 'content-type: application/json' -d @- https://testnet.ckbapp.dev/indexer
```

It returns empty objects.
```
{"jsonrpc":"2.0","result":{"last_cursor":"0x60c5e5dcf215925f7ef4dfaf5f4b4f105bc321c02776d6e7d52a1db3fcd9d011a401200000004580e3fd3a6623eb26f229239286cee63e18bcafb
38bc6c5d0de5a8c587647c200000000001a67ac000000020000002e","objects":[]},"id":2}
```

When I changed the `0x64` to `0x640`, there is a result.

```
{"jsonrpc":"2.0","result":{"last_cursor":"0x60c5e5dcf215925f7ef4dfaf5f4b4f105bc321c02776d6e7d52a1db3fcd9d011a401200000004580e3fd3a6623eb26f229239286cee63e18bcafb38bc6c5d0de5a8c587647c200000000001a6b970000000100000000","objects":[{"block_number":"0x1a6966","out_point":{"index":"0x0","tx_hash":"0x6cd8a1029222e9f4988306d852a33507350ed03b6e70c4b411e2268c0bdcec4e"},"output":{"capacity":"0x6fc23ac00","lock":{"args":"0x838e79dcef9cbf819a32778c8cfcc81bb2555561","code_hash":"0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8","hash_type":"type"},"type":{"args":"0x4580e3fd3a6623eb26f229239286cee63e18bcafb38bc6c5d0de5a8c587647c2","code_hash":"0xc5e5dcf215925f7ef4dfaf5f4b4f105bc321c02776d6e7d52a1db3fcd9d011a4","hash_type":"type"}},"output_data":"0x01000000000000000000000000000000","tx_index":"0x2"},{"block_number":"0x1a6997","out_point":{"index":"0x0","tx_hash":"0xd73c83b59f6a81dd9ef64ea56b97c0acb8d249c9e8e46a5dadf9ade1f620caed"},"output":{"capacity":"0x6fc23ac00","lock":{"args":"0x838e79dcef9cbf819a32778c8cfcc81bb2555561","code_hash":"0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8","hash_type":"type"},"type":{"args":"0x4580e3fd3a6623eb26f229239286cee63e18bcafb38bc6c5d0de5a8c587647c2","code_hash":"0xc5e5dcf215925f7ef4dfaf5f4b4f105bc321c02776d6e7d52a1db3fcd9d011a4","hash_type":"type"}},"output_data":"0x01000000000000000000000000000000","tx_index":"0x3"},{"block_number":"0x1a69d4","out_point":{"index":"0x0","tx_hash":"0x939d1fc87950bc818d9eee924bde48c61215a1391f03273b260fecb6086a616f"},"output":{"capacity":"0x6fc23ac00","lock":{"args":"0x838e79dcef9cbf819a32778c8cfcc81bb2555561","code_hash":"0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8","hash_type":"type"},"type":{"args":"0x4580e3fd3a6623eb26f229239286cee63e18bcafb38bc6c5d0de5a8c587647c2","code_hash":"0xc5e5dcf215925f7ef4dfaf5f4b4f105bc321c02776d6e7d52a1db3fcd9d011a4","hash_type":"type"}},"output_data":"0x01000000000000000000000000000000","tx_index":"0x1"}]},"id":2}
```

**Comment by quake** [comment]
could you check the `ckb-indexer` service version? I think this bug has been fixed in 0.2.0

**Comment by quake** [comment]
close stale issue
